canonical-web-and-design/maas-ui
# Remove power fields meant for individual nodes from add chassis form.

## Done
- Power type fields with `scope: "node"` are now filtered from add chassis form.
- Renamed `trimPowerParameters` to `formatPowerParameters` - now it acts as a bit of an integration layer between Formik and the add machine/add chassis apis (scanned the maas code and found that the add chassis api expects different parameters names than add machine).
- Added more tests to do with adding chassis.

## QA
- Click Add hardware > Chassis and check that the power types are filtered by `chassis: true` and the fields are filtered by `scope: "bmc"` according to `state.general.powerTypes.data`.
- Try to add chassis using the form fields given - note there is a [bug in the api](https://bugs.launchpad.net/maas/+bug/1876855) where some of the power types have `chassis: true` when they shouldn't be so you might get an error that the power type is not for chassis.

## Fixes
Fixes #1084

## Launchpad Issue
[1876680](https://bugs.launchpad.net/maas/+bug/1876680)
